components/ActionSheet/src/private/MDCActionSheetHeaderView.m

```diff
@@ -120,20 +121,26 @@ - (NSString *)title {
 
 - (void)setMessage:(NSString *)message {
   _messageLabel.text = message;
-  // If message is empty or nil then the title label's alpha value should be lighter, if there is both
-  // then the title label's alpha should be darker.
-  if (self.message && ![self.message isEqualToString:@""]) {
-    _titleLabel.alpha = TitleLabelAlpha;
-  } else {
-    _titleLabel.alpha = MessageLabelAlpha;
-  }
+  [self styleTitleAndMessage];
   [self setNeedsLayout];
 }
 
 - (NSString *)message {
   return _messageLabel.text;
 }
 
+- (void)styleTitleAndMessage {
+  // If message is empty or nil then the title label's alpha value should be lighter and primary
+  // color, if there is both then the title label's alpha should be darker secondary color.
+  if (self.message && ![self.message isEqualToString:@""]) {
+    _titleLabel.alpha = TitleLabelAlpha;
+  } else {
+    _titleLabel.alpha = MessageLabelAlpha;
+  }
+  _messageLabel.textColor = self.messageTextColor;
```

Comment:
I'm nervous that this logic may be somewhat unexpected from a client's point of view. I understand this may be a design request, but can we push back on this in a parallel thread so that we can simplify this behavior?

Reply:
I don't think clients will have an issue with this, I may be missing something. The `alpha` value is only changing if there is one label. This makes for a better UX. Again, I may be missing information as to what clients expect.

Reply:
My concern here is mainly in that we are conditionally changing the color values for the title and that if someone had set a specific alpha on their title color that the behavior here will be surprising.

Reply:
Okay do you feel like instead of touching the `alpha` for both labels you'd prefer we use `[UIColor.blackColor colorWithAlphaComponent:MessageLabelAlpha | TitleLabelAlpha];`?

Reply:
Let's be very careful with accessibility here. My understanding of `onSurfaceColor` is that it's a color that has sufficient accessibility for small text when place on top of `surfaceColor`. If that color has a contrast ratio of 4.5:1 (the minimum for WCAG AA small text) and we modify the alpha to make it lower - the text will no longer have sufficient contrast.

In general, I believe our themers (should) take into account the fact that contrast ratios may be right at the limit, and we should use them to apply the correct alphas for the colors in the scheme.